client/blocks/payment-request/index.js

```diff
@@ -39,6 +39,10 @@ const paymentRequestPaymentMethod = {
 			return false;
 		}
 
+		if ( ! getStripeServerData()?.shouldShowPaymentRequestButton ) {
+			return false;
+		}
+
```

Comment:
Sorry, I did not notice this before, but this doesn't look necessary 😅.

The PRB won't show up on the block cart/checkout because `wc_stripe_payment_request_params` won't be available due to [this line](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/0978b203a16f69406652d7f1465cb9720e7fc3fb/includes/payment-methods/class-wc-stripe-payment-request.php#L623).

We can revert the changes in `includes/class-wc-stripe-blocks-support.php` and revert `should_show_payment_button_on_cart` back to `private`.

What do you think?

Reply:
@ricardo I've refactored this bit significantly and rebased the branch on the latest release. It'd be fantastic if you could take nother look 🙏